numpy/lib/__init__.py

```diff
@@ -25,27 +25,29 @@
 from . import _function_base_impl
 from . import _stride_tricks_impl
 from . import _shape_base_impl
-from . import stride_tricks
 from . import _twodim_base_impl
 from . import _ufunclike_impl
 from . import _histograms_impl
 from . import _utils_impl
 from . import _arraysetops_impl
 from . import _polynomial_impl
 from . import _npyio_impl
-from . import npyio
-from . import arrayterator
+from . import _arrayterator_impl
 from . import _arraypad_impl
-from . import array_utils
 from . import _version
 
-from .arrayterator import Arrayterator
+# numpy.lib namespace members
+from ._arrayterator_impl import Arrayterator
 from ._version import *
 from .introspect import *
 from numpy._core._multiarray_umath import add_docstring, tracemalloc_domain
 from numpy._core.function_base import add_newdoc
 
-__all__ = ['emath']
```

Comment:
Searching on github I don't find any usages of `np.lib.emath`. Still, It's been here [for a long time](https://github.com/numpy/numpy/commit/30581a54d96448422cc550143e2b6fb860ef5c45) and we're still publicly exposing `np.emath`, so it would be nice to generate a nicer `AttributeError` with a migration in the `__getattr__` for the module.

Reply:
Sure! I added a proper error message for `np.lib.emath`.